objects/SetupChecker/script.lua

```diff
@@ -1880,8 +1880,56 @@ function randomSpirit(player)
     sourceSpirit.call("PickSpirit", {obj = spirit, color = player.color, aspect = "Random"})
     Player[player.color].broadcast("Your randomized spirit is "..spirit.getName(), "Blue")
 end
+function getGainSpiritChoices(obj)
+    local choices = {}
+    getXml(obj, {matchRecurse("GainSpirits", function (t)
+        for _, button in pairs(t.children) do
+            if button.tag == "Button" then
+                table.insert(choices, {spirit = button.attributes.spirit, aspect = button.attributes.aspect})
+            else
+                table.insert(choices, {})
+            end
+        end
+    end)})
+    return choices
+end
+function setupGainSpiritChoices(obj, choices)
+    local buttons = {}
+    for _, choice in pairs(choices) do
+        if choice.spirit ~= nil then
+            local label = choice.spirit
+            if choice.aspect ~= nil and choice.aspect ~= "" then
+                label = label .. " - " .. choice.aspect
+            end
+            table.insert(buttons, {
+                tag = "Button",
+                attributes = {
+                    id = "gain! " .. choice.spirit,
```

Comment:
is this id used for anything?

Reply:
the attributes are used by `getGainSpiritChoices` right? i'm not seeing the id field used there

Reply:
`pickGainSpiritChoice` is passed the id of the button pushed as an argument. It uses that to access the other attributes.

Reply:
ah hmm, i wonder if using the guid of the spirit might be better? i know sometimes folks use multiple copies of a spirit, though the gain spirit flow in general doesn't really support this fully